Problems building with target PPC440

I am having troubles to build openblas targeting the powerpc440. The compilation always fails  builiding the gemv_n.S file and the error says that there is junk at the end of some lines (I checked them  and there is nothing too weird in the code). 
Also I note that  in the kernel/power there are files that end with ppc440 (like gemv_n_ppc440.S), could be something related to that? 

The command I am running to build is:
make CC=powerpc-linux-gnu-gcc FC=powerpc-linux-gnu-gfortran HOSTCC=gcc-7 TARGET=PPC440

Just in case, I also tried to put the BINARY=32, and it also fails.  Also, if I change the compiler to powerpc64-linux-gnu-gcc the compilation finish, but the 64 bit version dont work for me.

```
../kernel/power/gemv_n.S: Assembler messages:
../kernel/power/gemv_n.S:433: Error: junk at end of line: `14'
../kernel/power/gemv_n.S:478: Error: junk at end of line: `15'
../kernel/power/gemv_n.S:523: Error: junk at end of line: `16'
../kernel/power/gemv_n.S:568: Error: junk at end of line: `17'
../kernel/power/gemv_n.S:613: Error: junk at end of line: `18'
../kernel/power/gemv_n.S:658: Error: junk at end of line: `19'
../kernel/power/gemv_n.S:703: Error: junk at end of line: `20'
../kernel/power/gemv_n.S:748: Error: junk at end of line: `21'
../kernel/power/gemv_n.S:780: Error: junk at end of line: `23'
../kernel/power/gemv_n.S:960: Error: junk at end of line: `14'
../kernel/power/gemv_n.S:961: Error: junk at end of line: `15'
../kernel/power/gemv_n.S:962: Error: junk at end of line: `16'
../kernel/power/gemv_n.S:963: Error: junk at end of line: `17'
../kernel/power/gemv_n.S:1130: Error: junk at end of line: `18'
../kernel/power/gemv_n.S:1131: Error: junk at end of line: `19'
../kernel/power/gemv_n.S:1132: Error: junk at end of line: `20'
../kernel/power/gemv_n.S:1133: Error: junk at end of line: `21'
../kernel/power/gemv_n.S:1165: Error: junk at end of line: `23'
../kernel/power/gemv_n.S:1689: Error: junk at end of line: `14'
../kernel/power/gemv_n.S:1690: Error: junk at end of line: `15'
../kernel/power/gemv_n.S:1786: Error: junk at end of line: `16'
../kernel/power/gemv_n.S:1787: Error: junk at end of line: `17'
../kernel/power/gemv_n.S:1828: Error: junk at end of line: `23'
../kernel/power/gemv_n.S:2379: Error: junk at end of line: `14'
../kernel/power/gemv_n.S:2380: Error: junk at end of line: `15'
../kernel/power/gemv_n.S:2381: Error: junk at end of line: `23'
../kernel/power/gemv_n.S:2742: Error: junk at end of line: `14'
../kernel/power/gemv_n.S:2743: Error: junk at end of line: `23'
Makefile.L2:262: recipe for target 'sgemv_n.o' failed
make[1]: *** [sgemv_n.o] Error 1
make[1]: *** Waiting for unfinished jobs....
../kernel/power/gemv_t.S: Assembler messages:
../kernel/power/gemv_t.S:421: Error: junk at end of line: `24'
../kernel/power/gemv_t.S:618: Error: junk at end of line: `14'
../kernel/power/gemv_t.S:619: Error: junk at end of line: `15'
../kernel/power/gemv_t.S:620: Error: junk at end of line: `16'
../kernel/power/gemv_t.S:621: Error: junk at end of line: `17'
../kernel/power/gemv_t.S:800: Error: junk at end of line: `18'
../kernel/power/gemv_t.S:801: Error: junk at end of line: `19'
../kernel/power/gemv_t.S:805: Error: junk at end of line: `20'
../kernel/power/gemv_t.S:806: Error: junk at end of line: `21'
../kernel/power/gemv_t.S:1588: Error: junk at end of line: `24'
../kernel/power/gemv_t.S:1781: Error: junk at end of line: `14'
../kernel/power/gemv_t.S:1782: Error: junk at end of line: `15'
../kernel/power/gemv_t.S:1786: Error: junk at end of line: `16'
../kernel/power/gemv_t.S:1787: Error: junk at end of line: `17'
../kernel/power/gemv_t.S:2219: Error: junk at end of line: `24'
../kernel/power/gemv_t.S:2340: Error: junk at end of line: `14'
../kernel/power/gemv_t.S:2341: Error: junk at end of line: `15'
../kernel/power/gemv_t.S:2617: Error: junk at end of line: `24'
../kernel/power/gemv_t.S:2726: Error: junk at end of line: `14'
Makefile.L2:265: recipe for target 'sgemv_t.o' failed
Makefile:170: recipe for target 'libs' failed```

Thanks for the quick response, what worked for me was to remove the ifeq, I could compile the source code and tested in the PPC440 that I have and all seems to work fine, thank you very much!

Forcing the AS compiler doesnt change the behavior, it trows the same "junk at the end of the line" error.
When I tried to use the ppc440 files trows the following error 
```
../kernel/power/gemv_n_ppc440.S: Assembler messages:
../kernel/power/gemv_n_ppc440.S:239: Error: invalid operands (*UND* and *ABS* sections) for `*'
Makefile.L2:262: recipe for target 'sgemv_n.o' failed
make[1]: *** [sgemv_n.o] Error 1
make[1]: *** Waiting for unfinished jobs....
../kernel/power/gemv_t_ppc440.S: Assembler messages:
../kernel/power/gemv_t_ppc440.S:231: Error: invalid operands (*UND* and *ABS* sections) for `*'
../kernel/power/gemv_t_ppc440.S:232: Error: invalid operands (*UND* and *ABS* sections) for `*'
Makefile.L2:265: recipe for target 'sgemv_t.o' failed
make[1]: *** [sgemv_t.o] Error 1
make[1]: Leaving directory '/home/seba/Documents/trabajo_dirigido/blas_stuffs/OpenBLAS/kernel'
Makefile:170: recipe for target 'libs' failed
make: *** [libs] Error 1
```

--- a/kernel/power/gemv_n_ppc440.S
+++ b/kernel/power/gemv_n_ppc440.S
@@ -131,6 +131,10 @@
 
 #define alpha  f27
 
+#if defined(PPC440)
+#define PREFETCHSIZE_A  (3 * 4)
+#endif
+
 #if defined(PPCG4)
 #define PREFETCHSIZE_A  (3 * 4)
 #endif

--- a/kernel/power/gemv_t_ppc440.S
+++ b/kernel/power/gemv_t_ppc440.S
@@ -96,6 +96,11 @@
 #define	X1	r22
 
 
+#if defined(PPC440)
+#define PREFETCHSIZE_A  42
+#define PREFETCHSIZE_C   7
+#endif
+
 #if defined(PPCG4)
 #define PREFETCHSIZE_A  42
 #define PREFETCHSIZE_C   7

